src/index.coffee

```diff
@@ -11,6 +11,6 @@ module.exports = (app, options) ->
   if !options or options.enabled isnt false
     autoMigrate = (options and options.migration) or 'auto-update'
     autoMigrate = require './' + autoMigrate
-    autoMigrate(app, options)
+    app.autoMigrateDone = autoMigrate(app, options).then -> debug "Done"
```

Comment:
After this comment https://github.com/strongloop/loopback/issues/418#issuecomment-283845301 I would also consider:
```
(app.components or= {}).autoMigrate = loaded: autoMigrate(app, options).then -> debug "loaded"`
```

Maybe the shy & easy start of a loopback's extension to components lifecycle :-)